### ISSUE TITLE
[BUG] pod for snapshot are stuck in terminating phase  ...

### ISSUE THREAD

**Issue by obeyler** [longhorn/longhorn#1793]
**Describe the bug**

so many of pod for snapshot are in terminating phase  ...
```
pvc-8a4913e7-6e25-47ee-af2a-a8de27991ec3-snap-c-1600254840zqwfp   0/1     Terminating   0          7h41m
pvc-8a4913e7-6e25-47ee-af2a-a8de27991ec3-snap-c-1600254840zqwkf   0/1     Terminating   0          8h
pvc-8a4913e7-6e25-47ee-af2a-a8de27991ec3-snap-c-1600254840zqzgd   0/1     Terminating   0          8h
pvc-8a4913e7-6e25-47ee-af2a-a8de27991ec3-snap-c-1600254840zr2cs   0/1     Terminating   0          7h3m
pvc-8a4913e7-6e25-47ee-af2a-a8de27991ec3-snap-c-1600254840zr4mw   0/1     Terminating   0          7h31m
pvc-8a4913e7-6e25-47ee-af2a-a8de27991ec3-snap-c-1600254840zr4v8   0/1     Terminating   0          8h
pvc-8a4913e7-6e25-47ee-af2a-a8de27991ec3-snap-c-1600254840zr5kz   0/1     Terminating   0          6h36m
pvc-8a4913e7-6e25-47ee-af2a-a8de27991ec3-snap-c-1600254840zr5ts   0/1     Terminating   0          7h54m
pvc-8a4913e7-6e25-47ee-af2a-a8de27991ec3-snap-c-1600254840zr7d2   0/1     Terminating   0          7h53m
pvc-8a4913e7-6e25-47ee-af2a-a8de27991ec3-snap-c-1600254840zr7ms   0/1     Terminating   0          6h49m
pvc-8a4913e7-6e25-47ee-af2a-a8de27991ec3-snap-c-1600254840zrd2h   0/1     Terminating   0          7h14m
pvc-8a4913e7-6e25-47ee-af2a-a8de27991ec3-snap-c-1600254840zrdfp   0/1     Terminating   0          7h55m
pvc-8a4913e7-6e25-47ee-af2a-a8de27991ec3-snap-c-1600254840zrf2t   0/1     Terminating   0          7h32m
pvc-8a4913e7-6e25-47ee-af2a-a8de27991ec3-snap-c-1600254840zrfm7   0/1     Terminating   0          7h43m
pvc-8a4913e7-6e25-47ee-af2a-a8de27991ec3-snap-c-1600254840zrg2n   0/1     Terminating   0          7h12m
pvc-8a4913e7-6e25-47ee-af2a-a8de27991ec3-snap-c-1600254840zrj8q   0/1     Terminating   0          7h3m
pvc-8a4913e7-6e25-47ee-af2a-a8de27991ec3-snap-c-1600254840zrl4g   0/1     Terminating   0          7h31m
pvc-8a4913e7-6e25-47ee-af2a-a8de27991ec3-snap-c-1600254840zrlx8   0/1     Terminating   0          8h
pvc-8a4913e7-6e25-47ee-af2a-a8de27991ec3-snap-c-1600254840zrn9l   0/1     Terminating   0          7h26m
pvc-8a4913e7-6e25-47ee-af2a-a8de27991ec3-snap-c-1600254840zrprj   0/1     Terminating   0          8h
pvc-8a4913e7-6e25-47ee-af2a-a8de27991ec3-snap-c-1600254840zrswk   0/1     Terminating   0          7h34m
pvc-8a4913e7-6e25-47ee-af2a-a8de27991ec3-snap-c-1600254840zrtrp   0/1     Terminating   0          7h59m
pvc-8a4913e7-6e25-47ee-af2a-a8de27991ec3-snap-c-1600254840zrv87   0/1     Terminating   0          7h37m
pvc-8a4913e7-6e25-47ee-af2a-a8de27991ec3-snap-c-1600254840zrvn8   0/1     Terminating   0          6h50m
pvc-8a4913e7-6e25-47ee-af2a-a8de27991ec3-snap-c-1600254840zrxbt   0/1     Terminating   0          7h53m
pvc-8a4913e7-6e25-47ee-af2a-a8de27991ec3-snap-c-1600254840zs2bt   0/1     Terminating   0          6h50m
pvc-8a4913e7-6e25-47ee-af2a-a8de27991ec3-snap-c-1600254840zs9zn   0/1     Terminating   0          7h22m
pvc-8a4913e7-6e25-47ee-af2a-a8de27991ec3-snap-c-1600254840zsbq6   0/1     Terminating   0          8h
pvc-8a4913e7-6e25-47ee-af2a-a8de27991ec3-snap-c-1600254840zsdk2   0/1     Terminating   0          7h18m
pvc-8a4913e7-6e25-47ee-af2a-a8de27991ec3-snap-c-1600254840zsg2t   0/1     Terminating   0          7h54m
pvc-8a4913e7-6e25-47ee-af2a-a8de27991ec3-snap-c-1600254840zsgrr   0/1     Terminating   0          8h
pvc-8a4913e7-6e25-47ee-af2a-a8de27991ec3-snap-c-1600254840zskfs   0/1     Terminating   0          7h58m
pvc-8a4913e7-6e25-47ee-af2a-a8de27991ec3-snap-c-1600254840zsksj   0/1     Terminating   0          7h42m
pvc-8a4913e7-6e25-47ee-af2a-a8de27991ec3-snap-c-1600254840zslqw   0/1     Terminating   0          7h20m
pvc-8a4913e7-6e25-47ee-af2a-a8de27991ec3-snap-c-1600254840zsmww   0/1     Terminating   0          6h51m
pvc-8a4913e7-6e25-47ee-af2a-a8de27991ec3-snap-c-1600254840zsqgp   0/1     Terminating   0          7h38m
pvc-8a4913e7-6e25-47ee-af2a-a8de27991ec3-snap-c-1600254840zsrdd   0/1     Terminating   0          6h32m
pvc-8a4913e7-6e25-47ee-af2a-a8de27991ec3-snap-c-1600254840zsrk5   0/1     Terminating   0          7h30m
pvc-8a4913e7-6e25-47ee-af2a-a8de27991ec3-snap-c-1600254840zss4g   0/1     Terminating   0          7h20m
pvc-8a4913e7-6e25-47ee-af2a-a8de27991ec3-snap-c-1600254840zsswb   0/1     Terminating   0          7h42m
pvc-8a4913e7-6e25-47ee-af2a-a8de27991ec3-snap-c-1600254840zsw2v   0/1     Terminating   0          6h45m
pvc-8a4913e7-6e25-47ee-af2a-a8de27991ec3-snap-c-1600254840zt4bj   0/1     Terminating   0          8h
pvc-8a4913e7-6e25-47ee-af2a-a8de27991ec3-snap-c-1600254840zt7n7   0/1     Terminating   0          6h59m
pvc-8a4913e7-6e25-47ee-af2a-a8de27991ec3-snap-c-1600254840zt8hd   0/1     Terminating   0          8h
pvc-8a4913e7-6e25-47ee-af2a-a8de27991ec3-snap-c-1600254840ztb4s   0/1     Terminating   0          7h50m
pvc-8a4913e7-6e25-47ee-af2a-a8de27991ec3-snap-c-1600254840ztcg5   0/1     Terminating   0          8h
pvc-8a4913e7-6e25-47ee-af2a-a8de27991ec3-snap-c-1600254840ztcgl   0/1     Terminating   0          7h6m
pvc-8a4913e7-6e25-47ee-af2a-a8de27991ec3-snap-c-1600254840ztfpt   0/1     Terminating   0          8h
pvc-8a4913e7-6e25-47ee-af2a-a8de27991ec3-snap-c-1600254840ztlnq   0/1     Terminating   0          6h48m
....
```
```
kubectl get pods -n longhorn-system |grep Terminating |wc -l
55427

```

**To Reproduce**
I use this storageclass:
```
kind: StorageClass
apiVersion: storage.k8s.io/v1
metadata:
  name: longhorn-1instance-backuped
  labels:
    app.kubernetes.io/instance: longhorn
    app.kubernetes.io/name: longhorn
    app.kubernetes.io/version: v1.0.2

provisioner: driver.longhorn.io
parameters:
  baseImage: ''
  dataLocality: best-effort
  fromBackup: ''
  numberOfReplicas: '1'
  recurringJobs: >-
    [ { "name":"snap", "task":"snapshot", "cron":"*/1 * * * *", "retain":1 }, {
    "name":"backup", "task":"backup", "cron":"0 0 * * *", "retain":1, "labels":
    { "interval":"1Day" } } ]
  staleReplicaTimeout: '30'
reclaimPolicy: Delete
volumeBindingMode: Immediate
```
**Expected behavior**

**Log**

```
time="2020-09-16T19:38:35Z" level=warning msg="Dropping Longhorn replica longhorn-system/pvc-529d7647-a4e9-4cef-ac07-ae6ae5f5919f-r-8f039bd1 out of the queue: fail to sync replica for longhorn-system/pvc-529d7647-a4e9-4cef-ac07-ae6ae5f5919f-r-8f039bd1: failed to get process pvc-529d7647-a4e9-4cef-ac07-ae6ae5f5919f-r-8f039bd1: rpc error: code = Unavailable desc = all SubConns are in TransientFailure, latest connection error: connection error: desc = \"transport: Error while dialing dial tcp 10.244.48.9:8500: connect: no route to host\""
time="2020-09-16T19:41:28Z" level=info msg="Longhorn replica longhorn-system/pvc-486b33ab-010f-454e-8f39-1e10b0496b3c-r-4a9d45cf has been deleted"
time="2020-09-16T19:42:33Z" level=debug msg="Replica controller b79262d6-7724-4416-9f5b-0bc884b6a9e4.k8s picked up pvc-8e56dfa3-f6e8-41ec-983e-718958147599-r-c94c2bbf"
time="2020-09-16T19:42:33Z" level=debug msg="Instance handler updated instance pvc-8e56dfa3-f6e8-41ec-983e-718958147599-r-c94c2bbf state, old state , new state stopped"
time="2020-09-16T19:42:33Z" level=info msg="Longhorn replica longhorn-system/pvc-8e56dfa3-f6e8-41ec-983e-718958147599-r-87f4de59 has been deleted"
time="2020-09-16T19:43:07Z" level=info msg="Longhorn replica longhorn-system/pvc-529d7647-a4e9-4cef-ac07-ae6ae5f5919f-r-8f039bd1 has been deleted"
time="2020-09-16T19:46:34Z" level=debug msg="Replica controller b79262d6-7724-4416-9f5b-0bc884b6a9e4.k8s picked up pvc-486b33ab-010f-454e-8f39-1e10b0496b3c-r-c65e09c4"
time="2020-09-16T19:46:34Z" level=debug msg="Instance handler updated instance pvc-486b33ab-010f-454e-8f39-1e10b0496b3c-r-c65e09c4 state, old state , new state stopped"
time="2020-09-16T19:46:34Z" level=info msg="Longhorn replica longhorn-system/pvc-486b33ab-010f-454e-8f39-1e10b0496b3c-r-c2302d0e has been deleted"
E0916 19:46:37.786001       1 replica_controller.go:177] fail to sync replica for longhorn-system/pvc-486b33ab-010f-454e-8f39-1e10b0496b3c-r-c65e09c4: failed to get process pvc-486b33ab-010f-454e-8f39-1e10b0496b3c-r-c65e09c4: rpc error: code = Unavailable desc = all SubConns are in TransientFailure, latest connection error: connection error: desc = "transport: Error while dialing dial tcp 10.244.48.9:8500: connect: no route to host"
time="2020-09-16T19:46:37Z" level=warning msg="Dropping Longhorn replica longhorn-system/pvc-486b33ab-010f-454e-8f39-1e10b0496b3c-r-c65e09c4 out of the queue: fail to sync replica for longhorn-system/pvc-486b33ab-010f-454e-8f39-1e10b0496b3c-r-c65e09c4: failed to get process pvc-486b33ab-010f-454e-8f39-1e10b0496b3c-r-c65e09c4: rpc error: code = Unavailable desc = all SubConns are in TransientFailure, latest connection error: connection error: desc = \"transport: Error while dialing dial tcp 10.244.48.9:8500: connect: no route to host\""
time="2020-09-16T19:47:33Z" level=info msg="Longhorn replica longhorn-system/pvc-8e56dfa3-f6e8-41ec-983e-718958147599-r-c94c2bbf has been deleted"
```

**Environment:**
 - Longhorn version: 1.0.2
 - Kubernetes version: 1.18
 - Node OS type and version: ubuntu

**Additional context**

**Comment by shuo-wu** [comment]
It seems that something is wrong with the `instance-manager-r-xxxxxx` in your Longhorn system. And what's the status of the related replicas?

**Comment by obeyler** [comment]
The replica is always attached detach attached detach ...

**Comment by obeyler** [comment]
I think  this issue is due to https://github.com/longhorn/longhorn/issues/1793
When I purge all terminating the 405 error disapear
so on I close this issue

**Comment by obeyler** [comment]
This night, I purge all the 56000 Terminating pod, remove the snapshot scheduling from the storage class and also the snapshot schedule on the volume
This morning I've got  66000 Terminating pod again
log from longhorn instance manager 
```

time="2020-09-17T07:10:18Z" level=warning msg="Dropping Longhorn node longhorn-system/d1968cbc-22dd-4d86-8eb1-a58f967003f2.k8s out of the queue: fail to sync node for longhorn-system/d1968cbc-22dd-4d86-8eb1-a58f967003f2.k8s: failed to get Kubernetes server version: Get \"https://10.245.0.1:443/version?timeout=32s\": net/http: request canceled (Client.Timeout exceeded while awaiting headers)"
E0917 07:10:19.890059       1 node_controller.go:301] fail to sync node for longhorn-system/b79262d6-7724-4416-9f5b-0bc884b6a9e4.k8s: failed to get Kubernetes server version: Get "https://10.245.0.1:443/version?timeout=32s": net/http: request canceled (Client.Timeout exceeded while awaiting headers)
time="2020-09-17T07:10:19Z" level=warning msg="Dropping Longhorn node longhorn-system/b79262d6-7724-4416-9f5b-0bc884b6a9e4.k8s out of the queue: fail to sync node for longhorn-system/b79262d6-7724-4416-9f5b-0bc884b6a9e4.k8s: failed to get Kubernetes server version: Get \"https://10.245.0.1:443/version?timeout=32s\": net/http: request canceled (Client.Timeout exceeded while awaiting headers)"
E0917 07:10:40.368329       1 node_controller.go:301] fail to sync node for longhorn-system/14485fe0-fe19-40ea-94e7-b62aa196d43b.k8s: failed to get Kubernetes server version: Get "https://10.245.0.1:443/version?timeout=32s": context deadline exceeded (Client.Timeout exceeded while awaiting headers)
time="2020-09-17T07:10:40Z" level=warning msg="Dropping Longhorn node longhorn-system/14485fe0-fe19-40ea-94e7-b62aa196d43b.k8s out of the queue: fail to sync node for longhorn-system/14485fe0-fe19-40ea-94e7-b62aa196d43b.k8s: failed to get Kubernetes server version: Get \"https://10.245.0.1:443/version?timeout=32s\": context deadline exceeded (Client.Timeout exceeded while awaiting headers)"
E0917 07:10:47.280618       1 node_controller.go:301] fail to sync node for longhorn-system/0491bf1a-16bf-4429-a126-7da3a659bb18.k8s: failed to get Kubernetes server version: Get "https://10.245.0.1:443/version?timeout=32s": net/http: request canceled (Client.Timeout exceeded while awaiting headers)
```
As there is too many pod the K8S cluster is dead.

**Comment by obeyler** [comment]
```
kubectl get pods -n longhorn-system |grep -v Terminating
NAME                                                              READY   STATUS        RESTARTS   AGE
csi-attacher-5469cbbb78-6jvz2                                     1/1     Running       3          21h
csi-attacher-5469cbbb78-km6xf                                     1/1     Running       1          11h
csi-attacher-5469cbbb78-m79c2                                     1/1     Running       0          9h
csi-provisioner-85dc49c5c9-lz65c                                  1/1     Running       0          9h
csi-provisioner-85dc49c5c9-mnlc5                                  1/1     Running       0          11h
csi-provisioner-85dc49c5c9-zmjhr                                  1/1     Running       11         21h
csi-resizer-56fd9d6855-4m9bw                                      1/1     Running       2          21h
csi-resizer-56fd9d6855-6vsc4                                      1/1     Running       0          9h
csi-resizer-56fd9d6855-lkdvh                                      1/1     Running       9          21h
engine-image-ei-ee18f965-7xshr                                    1/1     Running       0          21h
engine-image-ei-ee18f965-8wfpr                                    1/1     Running       1          21h
engine-image-ei-ee18f965-9pwvx                                    1/1     Running       0          21h
engine-image-ei-ee18f965-hmfgr                                    1/1     Running       0          21h
engine-image-ei-ee18f965-jft5q                                    1/1     Running       0          21h
engine-image-ei-ee18f965-lbpf2                                    1/1     Running       0          21h
engine-image-ei-ee18f965-zx9qk                                    1/1     Running       0          21h
instance-manager-e-1f8b8ad8                                       1/1     Running       0          21h
instance-manager-e-41f59e94                                       1/1     Running       0          21h
instance-manager-e-4a271fab                                       1/1     Running       0          9h
instance-manager-e-5de62d68                                       1/1     Running       0          21h
instance-manager-e-6837a4d2                                       1/1     Running       0          9h
instance-manager-e-9faa21cf                                       1/1     Running       0          21h
instance-manager-e-a079dffd                                       1/1     Running       0          21h
instance-manager-r-4221878a                                       1/1     Running       0          9h
instance-manager-r-5d33e1e7                                       1/1     Running       0          21h
instance-manager-r-a9cdf82a                                       1/1     Running       0          9h
longhorn-csi-plugin-4f4wm                                         2/2     Running       0          21h
longhorn-csi-plugin-fpjsb                                         2/2     Running       0          21h
longhorn-csi-plugin-hk482                                         2/2     Running       0          21h
longhorn-csi-plugin-ljdkc                                         2/2     Running       2          21h
longhorn-csi-plugin-pjfhf                                         2/2     Running       0          21h
longhorn-csi-plugin-rkrhj                                         2/2     Running       0          21h
longhorn-csi-plugin-xnxgn                                         2/2     Running       0          21h
longhorn-driver-deployer-d6468464b-kgx2r                          1/1     Running       0          9h
longhorn-manager-7ppdn                                            1/1     Running       3          21h
longhorn-manager-82mbx                                            1/1     Running       5          21h
longhorn-manager-ck5j9                                            1/1     Running       2          21h
longhorn-manager-djmzz                                            1/1     Running       5          21h
longhorn-manager-r997g                                            1/1     Running       3          21h
longhorn-manager-x9v87                                            1/1     Running       2          21h
longhorn-manager-zsgwz                                            1/1     Running       3          21h
longhorn-ui-cd8d4c9d7-5z4rg                                       1/1     Running       0          9h
pvc-07ee6f8e-0ad1-4b5a-ba44-b602202f405b-backup-c-160030087cxft   0/1     Pending       0          6m
pvc-486b33ab-010f-454e-8f39-1e10b0496b3c-c-f54ajj-c-16003158j5p   0/1     Completed     0          170m
pvc-486b33ab-010f-454e-8f39-1e10b0496b3c-c-f54ajj-c-160031l6svf   0/1     Completed     0          3h20m
pvc-486b33ab-010f-454e-8f39-1e10b0496b3c-c-f54ajj-c-160032pjgmg   0/1     Completed     0          140m
pvc-486b33ab-010f-454e-8f39-1e10b0496b3c-c-f54ajj-c-160032vm2b5   1/1     Running       0          110m
pvc-529d7647-a4e9-4cef-ac07-ae6ae5f5919f-c-3by76o-c-1600288nh7d   1/1     Running       0          12h
pvc-8e56dfa3-f6e8-41ec-983e-718958147599-c-9tceoi-c-1600284qw77   0/1     Completed     0          11h
pvc-8e56dfa3-f6e8-41ec-983e-718958147599-c-9tceoi-c-160029dzhlk   1/1     Running       0          9h

```

**Comment by joshimoo** [comment]
Thank you for reporting this, @khushboo-rancher are we able to reproduce this issue on a test cluster?

**Comment by obeyler** [comment]
@joshimoo @shuo-wu 
I see this morning that backup set by storage class launch lot of backup (45 backups instead of One)  (it should be a bug ?)
```
...   
     recurringJobs: >-
          [  {
          "name":"backup", "task":"backup", "cron":"0 0 * * *", "retain":1,
          "labels": { "interval":"1Day" } } ]
...
```

![image](https://user-images.githubusercontent.com/23224162/93578809-1030bd00-f99e-11ea-86e0-817712c1ca72.png)
witch can explain why when the previous storageclass was used with this parameter it crashed
```
  recurringJobs: >-
    [ { "name":"snap", "task":"snapshot", "cron":"*/1 * * * *", "retain":1 }, {
    "name":"backup", "task":"backup", "cron":"0 0 * * *", "retain":1, "labels":
    { "interval":"1Day" } } 
```

**Comment by khushboo-rancher** [comment]
I tried with the same below storage class but couldn't reproduce the issue. Looks like there could be a disk space shortage in the node to which the volumes are attached.
```
kind: StorageClass
apiVersion: storage.k8s.io/v1
metadata:
  name: longhorn-1instance-backuped
  labels:
    app.kubernetes.io/instance: longhorn
    app.kubernetes.io/name: longhorn
    app.kubernetes.io/version: v1.0.2

provisioner: driver.longhorn.io
parameters:
  baseImage: ''
  dataLocality: best-effort
  fromBackup: ''
  numberOfReplicas: '1'
  recurringJobs: >-
    [ { "name":"snap", "task":"snapshot", "cron":"*/1 * * * *", "retain":1 }, {
    "name":"backup", "task":"backup", "cron":"0 0 * * *", "retain":1, "labels":
    { "interval":"1Day" } } ]
  staleReplicaTimeout: '30'
reclaimPolicy: Delete
volumeBindingMode: Immediate
```

@obeyler Could you please check your disk space in the node where the volumes are attached if it has enough space to perform the creation of a new snapshot and deletion of the old one?

**Comment by obeyler** [comment]
We have no pb with disk space neither on dedicated node for persistence nor in S3
our K8S topology: 
external etcd cluster (3 dedicated VM)
3 node master
4 node dedicated to persistence (taint node=persistence)
3 node without taint
2 node dedicated to be exposed (taint node=public)

**Comment by joshimoo** [comment]
This looks like it has the same root cause as #1903 
This happens because we set NodeName in the pod spec, which forces kubernetes to schedule the pod onto that node.
Then the taint controller removes it afterwards since it does not have the correct tolerations.

https://github.com/longhorn/longhorn-manager/blob/b9e8a3d05c93aa1801090482c14b9dd63c546e3a/controller/volume_controller.go#L1941

@khushboo-rancher to reproduce create a dedicated storage node, with a taint, then allow that taint via the longhorn settings.
The jobs will be created on that node, but then killed by the taint controller, which will start the jobs all over again.

**Comment by yasker** [comment]
#1796 is going to remove the nodeName from the recurring jobs.

**Comment by yasker** [comment]
This issue should be automatically resolved once we resolved #1796 and #1903

**Comment by yasker** [comment]
Duplicate of #1903
